src/commands/functions/create.js

```diff
@@ -373,6 +367,42 @@ const createFunctionAddon = async function ({ api, addons, siteId, addonName, si
   }
 }
 
+const injectEnvVariables = async ({ context }) => {
+  const { log, warn, error, netlify } = context
+  const { api, site } = netlify
+  const { teamEnv, addonsEnv, siteEnv, dotFilesEnv } = await getSiteInformation({
+    api,
+    site,
+    warn,
+    error,
+  })
+  await addEnvVariables({ log, teamEnv, addonsEnv, siteEnv, dotFilesEnv })
+}
+
+const handleOnComplete = async ({ context, onComplete }) => {
+  if (onComplete) {
+    await injectEnvVariables({ context })
+    await onComplete.call(context)
+  }
+}
+
+const handleAddonDidInstall = async ({ addonCreated, addonDidInstall, context, fnPath }) => {
+  if (addonCreated && addonDidInstall) {
```

Comment:
[sand] Could we use early returns?
`if (!addonCreated || !addonDidInstall) { return }`
`if (!confirmPostInstall) { return }`